provider/execution/execution.go

```diff
@@ -39,10 +40,12 @@ type TableExecutor struct {
 	executionStart time.Time
 	// columns of table, this is to reduce calls to sift each time
 	columns [2]schema.ColumnList
+	// goroutinesSem to limit number of goroutines (resources fetched) concurrently
+	goroutinesSem *semaphore.Weighted
```

Comment:
nit:
```suggestion
	maxGoutines *semaphore.Weighted
```

Reply:
My idea behind the naming here was that the argument is `max_goroutines` (uint64) and the semaphore is `goroutinesSem` as it can be less than max depending on the flow.